compare_locales/resource/from_properties.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import re
+from typing import Iterable, List, Optional, Union
+
+from ..parsers import Comment as ParserComment
+from ..parsers import Entity as ParserEntity
+from ..parsers import Junk as ParserJunk
+from ..parsers import Whitespace as ParserWhitespace
+from .elements import (
+    Comment,
+    Entry,
+    Junk,
+    Message,
+    Pattern,
+    PatternMessage,
+    Text,
+    VariableRef,
+)
+
+
+def patternFromPropertiesValue(
+    value: str, variables: Optional[re.Pattern[str]] = None
+) -> PatternMessage:
+    """
+    Compile a .properties value into a PatternMessage data class.
+
+    `variables` may be a compiled regexp identifying a variable placeholder from the value.
+    Its captured groups (if any) are ignored.
+    """
+
+    pattern: Pattern = []
+    prevEnd = 0
+    if variables:
+        for match in variables.finditer(value):
+            start = match.start()
+            if start > prevEnd:
+                pattern.append(Text(value[prevEnd:start]))
+            pattern.append(VariableRef(match[0]))
+            prevEnd = match.end()
```

Comment:
could this result in `prevEnd` not always increasing?  I.e. should it be `prevEnd = max(prevEnd, match.end())`?

Reply:
Because `.finditer()` only returns non-overlapping matches left-to-right, using `max()` would not change the behaviour of this line. The only way for `match.end()` to not be greater than `prevEnd` is if the match is empty. Which is technically valid, but mostly nonsensical.